Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/approximate_convex_decomposition.h

```diff
@@ -0,0 +1,1689 @@
+// Copyright (c) 2025 GeometryFactory (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+// Author(s)     : Sven Oesau
+//
+
+#ifndef CGAL_POLYGON_MESH_PROCESSING_CONVEX_DECOMPOSITION_H
+#define CGAL_POLYGON_MESH_PROCESSING_CONVEX_DECOMPOSITION_H
+
+#include <CGAL/license/Polygon_mesh_processing/combinatorial_repair.h>
+
+#include <unordered_set>
+
+#include <CGAL/Iso_cuboid_3.h>
+
+#include <CGAL/Named_function_parameters.h>
+#include <CGAL/boost/graph/named_params_helper.h>
+
+#include <CGAL/convex_hull_3.h>
+#include <CGAL/Polygon_mesh_processing/bbox.h>
+#include <CGAL/Polygon_mesh_processing/compute_normal.h>
+#include <CGAL/Polygon_mesh_processing/triangle.h>
+#include <CGAL/Polygon_mesh_processing/shape_predicates.h>
+#include <CGAL/Polygon_mesh_processing/measure.h>
+#include <CGAL/utils_classes.h>
+
+#include <CGAL/Surface_mesh.h>
+#include <CGAL/IO/polygon_soup_io.h>
+
+#include <CGAL/AABB_tree.h>
+#include <CGAL/AABB_traits_3.h>
+#include <CGAL/AABB_face_graph_triangle_primitive.h>
+
+#ifdef CGAL_LINKED_WITH_TBB
+#include <atomic>
+#include <tbb/parallel_for_each.h>
+#include <tbb/concurrent_unordered_map.h>
+#include <tbb/concurrent_priority_queue.h>
+#else
+#include <unordered_map>
+#endif
+
+#include <CGAL/Timer.h>
+#include <CGAL/Memory_sizer.h>
+
+#include <queue>
+
+namespace CGAL {
+namespace Polygon_mesh_processing {
+namespace internal {
+
+using Vec3_uint = std::array<unsigned int, 3>;
+
+struct Bbox_uint {
+  Vec3_uint lower;
+  Vec3_uint upper;
+  Bbox_uint(const Vec3_uint &lower, const Vec3_uint &upper) : lower(lower), upper(upper) {}
+};
+
+enum Grid_cell : int8_t {
```

Comment:
We could as well use `CGAL::Bounded_side`.   Is it important that it is `int_8` because this is the type of a data member?